packages/cdktf/lib/terraform-resource-targets.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { TerraformResource } from "./terraform-resource";
+
+/**
+ *
+ */
+export class TerraformResourceTargets {
+  private _resourceTargetMap: Map<string, TerraformResource> = new Map<
+    string,
+    TerraformResource
+  >();
+
+  private prettyPrintEntries() {
+    return [...this._resourceTargetMap.entries()].reduce(
+      (accum, curr) => accum + "\n" + `${curr[0]}: ${curr[1].friendlyUniqueId}`,
+      ""
+    );
+  }
+
+  public addResourceTarget(resource: TerraformResource, target: string): void {
+    if (this._resourceTargetMap.has(target)) {
+      throw new Error(
+        `Target "${target}" has already been set at ${
+          this._resourceTargetMap.get(target)?.friendlyUniqueId
+        }`
+      );
+    }
+    this._resourceTargetMap.set(target, resource);
+  }
+
+  public getResourceByTarget(target: string): TerraformResource {
+    const result = this._resourceTargetMap.get(target);
+    if (!result) {
+      throw new Error(`Target "${target}" has not been set:
+      
+            Current Target Entries:\n ${this.prettyPrintEntries()}
+
+            To add this target, call .addMoveTarget("${target}") on the instance of the resource to move to.
```

Comment:
Can't help but to think in TS much of the time 😅. It is a bit of an annoying problem since making it language agnostic may make it more difficult to parse (i.e. using something like `add move target resource instance function with parameter "target"`). 

Maybe we can have something that detects the language being used and translates it to the form of that language (i.e. `.addMoveTarget("target")` -> `.add_move_target("target")` in the case of Python). Though that supposes the config is accessible where the error is being thrown. Just an idea 😁